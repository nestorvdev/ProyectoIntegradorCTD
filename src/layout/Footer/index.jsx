import React, {useState} from 'react';
import Styles from './styles.module.css';
import StylesApp from "../../App.module.css"
import logoFacebook from "./img/icon facebook.png";
import logoLinkedin from "./img/icon linkedin.png";
import logoTwitter from "./img/tweet.png";
import logoInstagram from "./img/icon ig.png"
import Share from '../../components/Product/Share';
import iconSocial from "./img/icon-social.svg"

function Footer({ showBurger }) {
    const [shareIsOpenFooter, setShareIsOpenFooter]= useState(false);
    const openShareModal = (() => { setShareIsOpenFooter(true)});
    let placeShareCall="footer"

    return (
        <footer className={`${Styles.footer} ${StylesApp.delimiter}`} >
            <div className={`${Styles.delimiterChild} ${StylesApp.delimiterChild} `}>
                <div>
                    <h3 className={Styles.footerTittle} >2021 @ World Guest Booking</h3>
                </div>
                <div className={Styles.icons}>
                    <a href="https://www.facebook.com/profile.php?id=100074545225635" target="_blank" rel="noreferrer"><img className={Styles.iconsImg} src={logoFacebook} alt="" /></a>
                    <a href="https://www.linkedin.com/in/world-guest-2b195b225/" target="_blank" rel="noreferrer"><img className={Styles.iconsImg} src={logoLinkedin} alt="" /></a>
                    <a href="https://twitter.com/WorldGuest2" target="_blank" rel="noreferrer"><img className={Styles.iconsImg} src={logoTwitter} alt="" /></a>
                    <a href="https://www.instagram.com/world_guest_booking/" target="_blank" rel="noreferrer"><img className={Styles.iconsImg} src={logoInstagram} alt="" /></a>
                    <img src={iconSocial} alt="iconSocial" className={`${Styles.iconsImg}`} onClick={openShareModal} />
                    <Share shareIsOpen={shareIsOpenFooter} placeShareCall={placeShareCall} setShareIsOpen={setShareIsOpenFooter} />
                </div>
            </div>
        </footer>
    );
}

export default Footer;