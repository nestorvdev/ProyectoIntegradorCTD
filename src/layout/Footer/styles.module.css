.footer {
  background-color: var(--primary-color);
  bottom: 0px;
  color: white;
  flex-wrap: nowrap;
  height: 58px;
  position: fixed;
  width: 100%;
  z-index: 200;
}

footer>.delimiterChild {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding-left: clamp(20px,3.9%,50px);
  padding-right: clamp(20px,3.9%,50px);
}

.footerTittle {
  font-size: var(--heading4);
  font-weight: var(--font-bold);
}

.iconsImg {
  margin: 0 8px;
  
}
@media all and (max-width: 550px) {
  .iconsImg {
    margin: 0 5px;
    width:20px
    
  }

  .footerTittle {
    font-size: 12pt;
  }

}

@media all and (max-width: 415px) {
  /*Footer */
  .footer {
    z-index: 99;
  }
  .icons {
    display: none;
  }
}
