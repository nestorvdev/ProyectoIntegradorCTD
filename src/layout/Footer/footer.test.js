

import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Footer from "./index";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <Footer/>", () => {
    let footer; 

    beforeEach(() => {
        footer = shallow(<Footer />)
    });

    it("Deberia mostrar <Footer/> correctamente", () => {
        expect(footer).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta h3", () => {
        let node = footer.find("h3")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 4 etiquetas img", () => {
        let node = footer.find("img")
        expect(node.length).toEqual(5)
    });
});