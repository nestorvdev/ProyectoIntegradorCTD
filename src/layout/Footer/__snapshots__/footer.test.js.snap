// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <Footer/> Deberia mostrar <Footer/> correctamente 1`] = `
<footer
  className="footer delimiter"
>
  <div
    className="delimiterChild delimiterChild "
  >
    <div>
      <h3
        className="footerTittle"
      >
        2021 @ World Guest Booking
      </h3>
    </div>
    <div
      className="icons"
    >
      <a
        href="https://www.facebook.com/profile.php?id=100074545225635"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          className="iconsImg"
          src="icon facebook.png"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/world-guest-2b195b225/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          className="iconsImg"
          src="icon linkedin.png"
        />
      </a>
      <a
        href="https://twitter.com/WorldGuest2"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          className="iconsImg"
          src="tweet.png"
        />
      </a>
      <a
        href="https://www.instagram.com/world_guest_booking/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          className="iconsImg"
          src="icon ig.png"
        />
      </a>
      <img
        alt="iconSocial"
        className="iconsImg"
        onClick={[Function]}
        src="icon-social.svg"
      />
      <Share
        placeShareCall="footer"
        setShareIsOpen={[Function]}
        shareIsOpen={false}
      />
    </div>
  </div>
</footer>
`;
